src/sema/assembly/expression.rs

```diff
@@ -207,3 +272,427 @@ fn resolve_string_literal(
 }
+
+fn resolve_variable_reference(
+    id: &pt::Identifier,
+    ns: &mut Namespace,
+    symtable: &Symtable,
+    context: &ExprContext,
+) -> Result<AssemblyExpression, ()> {
+    if let Some(v) = symtable.find(&id.name) {
+        match &v.usage_type {
+            VariableUsage::AssemblyLocalVariable => {
+                return Ok(AssemblyExpression::AssemblyLocalVariable(
+                    id.loc,
+                    v.ty.clone(),
+                    v.pos,
+                ))
+            }
+            VariableUsage::AnonymousReturnVariable => {
+                unreachable!("Anonymous returns variables cannot be accessed from assembly blocks")
+            }
+            _ => {
+                return Ok(AssemblyExpression::SolidityLocalVariable(
+                    id.loc,
+                    v.ty.clone(),
+                    v.storage_location.clone(),
+                    v.pos,
+                ))
+            }
+        }
+    }
+
+    match ns.resolve_var(context.file_no, context.contract_no, id, false) {
+        Some(Symbol::Variable(_, Some(var_contract_no), var_no)) => {
+            let var = &ns.contracts[*var_contract_no].variables[*var_no];
+            if var.constant {
+                Ok(AssemblyExpression::ConstantVariable(
+                    id.loc,
+                    var.ty.clone(),
+                    Some(*var_contract_no),
+                    *var_no,
+                ))
+            } else {
+                Ok(AssemblyExpression::StorageVariable(
+                    id.loc,
+                    var.ty.clone(),
+                    *var_contract_no,
+                    *var_no,
+                ))
+            }
+        }
+        Some(Symbol::Variable(_, None, var_no)) => {
+            let var = &ns.constants[*var_no];
+            Ok(AssemblyExpression::ConstantVariable(
+                id.loc,
+                var.ty.clone(),
+                None,
+                *var_no,
+            ))
+        }
+        None => {
+            ns.diagnostics.push(Diagnostic::decl_error(
+                id.loc,
+                format!("'{}' is not found", id.name),
+            ));
+            Err(())
+        }
+
+        _ => {
+            ns.diagnostics.push(Diagnostic::error(
+                id.loc,
+                "only variables can be accessed inside assembly blocks".to_string(),
+            ));
+            Err(())
+        }
+    }
+}
+
+fn resolve_function_call(
+    functions: &IndexMap<String, AssemblyFunction>,
+    func_call: &AssemblyFunctionCall,
+    context: &ExprContext,
+    symtable: &Symtable,
+    ns: &mut Namespace,
+) -> Result<AssemblyExpression, ()> {
+    if func_call.id.name.starts_with("verbatim") {
+        ns.diagnostics.push(Diagnostic::error(
+            func_call.id.loc,
+            "verbatim functions are not yet supported in Solang".to_string(),
+        ));
+        return Err(());
+    } else if assembly_unsupported_builtin(&func_call.id.name[..]) {
```

Comment:
This could be written as `assembly_unsupported_builtin(func_call.id.name.as_str())`. I guess this a question of taste though.